libpromises/locks.c

```diff
@@ -365,27 +365,91 @@ static int RemoveLock(const char *name)
     return 0;
 }
 
+static bool NoOrObsoleteLock(LockData *entry, ARG_UNUSED size_t val_size, size_t *max_old)
+{
+    assert((entry == NULL) || (val_size == sizeof(LockData)));
```

Comment:
This assertion is failing in Travis. AFAICT, for `MDB_NOTFOUND` this function is called with non-null `entry`, and 0 `val_size`.

Reply:
How would that be possible after these two lines?
```
    void *cur_val = (rc == MDB_SUCCESS ? orig_data.mv_data : NULL);
    size_t cur_val_size = (rc == MDB_SUCCESS ? orig_data.mv_size : 0);
```

Reply:
Yes, but you called the function with `cur_val_copy`, not `cur_val`.

This change, which you force pushed, fixed the issue:

https://github.com/cfengine/core/compare/28353e0f912d961f802f1f14fedc9354daf6ba58..5a1ccd73ada42e9e4bbb3e761409c3add253a4dc